### PR TITLE
Support for more filters

### PR DESCRIPTION
**Pull request by lpabon**
Add support for github.com/Masterminds/sprig in texts to support more filters

### CODE REVIEW

**Comment by lpabon** [comment]
Not sure why one run of travis failed, but the 1.10.x didn't.

File: renderer.go [hunk 1]
```diff
@@ -110,7 +112,7 @@ type textRenderer struct {
 }
 
 func (mr *textRenderer) Apply(template *Template) ([]byte, error) {
-	tmpl, err := text_template.New("Text Template").Funcs(funcMap).Parse(mr.inputTemplate)
+	tmpl, err := text_template.New("Text Template").Funcs(funcMap).Funcs(sprig.TxtFuncMap()).Parse(mr.inputTemplate)
```
[the remark this answers is not in the document]
Comment (lpabon) [reply renderer.go]: Good, Will do

**Comment by lpabon** [comment]
I also added an IsMap boolean.

File: renderer.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (lpabon) [reply renderer.go]: done

**Comment by lpabon** [comment]
This is what I am using it for: https://github.com/libopenstorage/libopenstorage.github.io

File: renderer.go [hunk 2]
```diff
@@ -128,7 +130,7 @@ type htmlRenderer struct {
 }
 
 func (mr *htmlRenderer) Apply(template *Template) ([]byte, error) {
-	tmpl, err := html_template.New("Text Template").Funcs(funcMap).Parse(mr.inputTemplate)
+	tmpl, err := html_template.New("Text Template").Funcs(funcMap).Funcs(sprig.TxtFuncMap()).Parse(mr.inputTemplate)
```
Comment (pseudomuto) [inline renderer.go]: /nit I know they both return `GenericFuncMap` [(source)](https://github.com/Masterminds/sprig/blob/6b2a58267f6a8b1dc8e2eb5519b984008fa85e8c/functions.go#L46-L53), but just to be clearer and protect against future changes, can we make this one use `sprig.HtmlFuncMap()`

File: template.go [hunk 3]
```diff
@@ -147,6 +147,7 @@ type MessageField struct {
 	Type         string `json:"type"`
 	LongType     string `json:"longType"`
 	FullType     string `json:"fullType"`
+	IsMap        bool   `json:"ismap"`
```
Comment (pseudomuto) [inline template.go]: I think we should add tests for this

**Comment by pseudomuto** [comment]
@lpabon are you still planning on adding some tests for the `IsMap` thing?

**Comment by lpabon** [comment]
Added unit tests and changed filter to sprig.HtmlFuncMap()

**Comment by lpabon** [comment]
Looks like master has travis tests have some other race condition errors. AFAIK, not caused by this PR.

**Pull request merged**